DOCS-3911 Browser Test Runs Edits and Settings Edit (#16522)
DataDog/documentation: content/en/synthetics/browser_tests/test_results.md

- link: "/synthetics/guide/explore-rum-through-synthetics/"
  tag: "Documentation"
  text: "Explore RUM & Session Replay in Synthetics"
- link: "/synthetics/dashboards/browser_test/"
  tag: "Documentation"
  text: "Learn about the Browser Test Performance Dashboard"
---

## Overview

> Test runs appear in a test details page after a Synthetic test executes. [Sample runs](#sample-runs) correlate to the latest passed and failed test executions over a time interval and in a specific number of locations and devices.

Reviewer: I don't see a sample runs section on this page. Perhaps you meant #sample-results?

Author: ```suggestion
Test runs appear in a test details page after a Synthetic test executes. [Sample results](#sample-results) correlate to the latest passed and failed test executions over a time interval and in a specific number of locations and devices.
```

## Test properties

In the **Properties** section, you can see the test ID, test creation and edit date, a list of tags, test priority, and a link to an out-of-the-box Synthetic [browser test dashboard][11].

**Overview** 

> : This section describes the test URL, the number of locations, the number of devices, the test interval, and the number of test steps, including custom steps.

Reviewer: ```suggestion
: This section describes the test URL, number of locations, number of devices, test interval, and the number of test steps, including custom steps.
```

**Monitor**
: This section contains the name of the [Synthetic test's monitor][13] and the configured notification message.

**CI/CD Execution**
: This section contains a dropdown menu to change the [execution rule][12] for this test running as part of a [Continuous Testing CI pipeline][19].

## Test history

In the **History** section, you can see three graphs:

- The **Global Uptime** graph displays the total uptime of all test locations in a given time interval. The global uptime takes into consideration the [alert conditions][20] configured for a test.

> - The **Time-to-interactive by location and device** graph displays the amount of time until a page could be interacted with in seconds. For more information about uptime monitoring, see the [Website Uptime Monitoring with SLOs][14] guide.

Reviewer: ```suggestion
- The **Time-to-interactive by location and device** graph displays the amount of time until a page can be interacted with in seconds. For more information about uptime monitoring, see the [Website Uptime Monitoring with SLOs][14] guide.
```

- The **Test duration by location and device** graph displays the amount of time in minutes each location and device takes to complete in a given time interval. 

{{< img src="synthetics/browser_tests/history.png" alt="The History and Sample Runs section in the Test Details page" style="width=80%" >}}

## Sample results

> Browser test runs include components such as [screenshots](#screenshots), [page performance data](#page-performance), [errors](#errors), [resources](#resources), and [backend traces](#backend-traces) to help troubleshoot your [test failure](#failed-test-runs).

Reviewer: ```suggestion
Browser test runs include components such as [screenshots](#screenshots-and-actions), [page performance data](#page-performance), [errors](#errors-and-warnings), [resources](#resources), and [backend traces](#backend-traces) to help troubleshoot your [test failure](#failed-results).
```

Reviewer: Looks like you changed your mind about what the headers eventually became? :)

: The type of browser your test was executed from.

Time ran
: The amount of time that has passed since your test ran.

Run type
: The type of test run (CI, fast retry, manually triggered, or scheduled).

### RUM sessions

To view related sessions and available replays in the RUM Explorer, click **View Session in RUM**. To access a user session for a particular action or step in Session Replay, click **Replay Session**. For more information, see [Explore RUM & Session Replay in Synthetics][16].

### Screenshots and actions

> Every executed test step contains a screenshot of the step action, a link to the session in Session Replay, the step description, the starting URL for a given step, the step ID, the step duration, and page performance information.

Reviewer: ```suggestion
Every executed test step contains a screenshot of the step action, a link to the session in Session Replay, the step description, starting URL for a given step, step ID, step duration, and page performance information.
```

### Page performance

> Synthetic Monitoring displays two [Core Web Vital metrics][6] ([Largest Contentful Paint][2] and [Cumulative Layout Shift][3]) as pills to the right of each step URL. [First Input Delay][4] is available if you are using [Real User Monitoring][5] to collect real user data. For more information, see [Monitoring Page Performance][6].

Author: ```suggestion
Synthetic Monitoring includes two [Core Web Vital metrics][6] ([Largest Contentful Paint][2] and [Cumulative Layout Shift][3]) as lab metrics, displayed as pills to the right of each step URL. [First Input Delay][4] is available as a real metric if you are using [Real User Monitoring][5] to collect real user data. For more information, see [Monitoring Page Performance][6].
```
```suggestion
Synthetic Monitoring displays two [Core Web Vital metrics][6] ([Largest Contentful Paint][2] and [Cumulative Layout Shift][3]) as pills to the right of each step URL. [First Input Delay][4] is available if you are using [Real User Monitoring][5] to collect real user data. For more information, see [Monitoring Page Performance][6].
```

### Errors and warnings

> Click the **Errors** pill to access the **Errors & Warnings** tab and examine a list of errors separated by error type (`js` or `network`) and status (the network status code).

Reviewer: Would suggest adding a screenshot that makes it clear what the "Error pill" looks like, as that isn't a UI/UX term I'm familiar with.

Protocol
: The protocol of the request.

Status
: The HTTP response status code.

Duration
: The time needed to perform the request.

Size
: The size of the request response.

### Backend traces

Click the **Traces** pill to access the **Traces** tab and explore APM traces associated with the browser test. While the UI is similar to the [Trace View][7] in the Trace Explorer, one browser test step can make multiple requests to different URLs or endpoints. This results in several associated traces, depending on your tracing setup and on the URLs you allowed in for browser tests in the [Synthetic Monitoring Settings page][8]. 

For more information about cross-product correlation, see the [Ease Troubleshooting With Cross-Product Correlation][21] guide.

### Step duration

The step duration represents the amount of time the step takes to execute with the [Datadog locator system][9]. Not only does the step duration include the action (such as user interactions), but also it incorporates the wait and retry mechanism, which allows browser tests to ensure an element is able to be interacted with. For more information, see [Advanced Options for Browser Test Steps][9].

## Failed results

> A test result is considered `FAILED` if it does not satisfy its assertions or if a step failed for another reason. You can troubleshoot failed runs by looking at their screenshots, checking for potential [errors](#errors) at the step level, and looking into [resources][17] and [backend traces](#backend-traces) generated by their steps.

Reviewer: ```suggestion
A test result is considered `FAILED` if it does not satisfy its assertions or if a step failed for another reason. You can troubleshoot failed runs by looking at their screenshots, checking for potential [errors](#errors-and-warnings) at the step level, and looking into [resources][17] and [backend traces](#backend-traces) generated by their steps.
```
